### PR TITLE
Runtime: add an explicit cast; NFC

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

This adds an explicit cast of the atomic type to a `void *` prior to comparing
it to nullptr.  This is generally unnecessary, however, the Windows C++ library
(msvcprt) from Visual Studio 2014 (WinSDK 10.0.10586.0) has extensions which
makes the conversion ambiguous.  Simply cast the value on all targets.

### CODE REVIEW

File: include/swift/Runtime/Metadata.h [hunk 1]
```diff
@@ -1874,10 +1874,10 @@ struct TargetForeignTypeMetadata : public TargetMetadata<Runtime> {
   setCachedUniqueMetadata(
     ConstTargetMetadataPointer<Runtime, swift::TargetForeignTypeMetadata> unique)
   const {
-    assert((asFullMetadata(this)->Unique == nullptr
```
Comment (gribozavr) [inline include/swift/Runtime/Metadata.h]: Does `!asFullMetadata(this)->Unique` work?

Comment (compnerd) [inline include/swift/Runtime/Metadata.h]: No :-(.  That results in an ambiguous conversion as well.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
@compnerd Could you take a look at the compilation error?

```
swift/include/swift/Runtime/Metadata.h:1877:13: error: cannot cast from type 'std::atomic<ConstTargetMetadataPointer<InProcess, swift::TargetForeignTypeMetadata> >' (aka 'atomic<const swift::TargetForeignTypeMetadata<swift::InProcess> *>') to pointer type 'void *'
    assert((static_cast<void *>(asFullMetadata(this)->Unique) == nullptr ||
```

**Comment by compnerd** [comment]
Ugh, thats slightly annoying, but managed to reproduce it locally on Linux as well.  I did a slightly more complicated fix that I think improves the code overall.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
@compnerd

```
swift/include/swift/Runtime/Metadata.h:1860:3: error: unknown type name 'RuntimeMetadataPointer'
  RuntimeMetadataPointer getCachedUniqueMetadata() const {
```

**Comment by compnerd** [comment]
Sorry, I had pushed the wrong version.  I had scoped the alias to the inner struct rather than the outer struct.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request merged**